pkg/vcs/linux.go

```diff
@@ -39,32 +39,54 @@ func newLinux(dir string, opts []RepoOpt) *linux {
 	}
 }
 
-func (ctx *linux) PreviousReleaseTags(commit string) ([]string, error) {
+func (ctx *linux) FetchRemote(repo string) error {
```

Comment:
I don't really see where it gets invoked.

Also, maybe we can consider some more descriptive name for the function?

Reply:
This is overwriting the generic [`git.fetchRemote`](https://github.com/google/syzkaller/blob/787ed7e0bfb3cf1da3025aff0c7770717a993667/pkg/vcs/git.go#L136) which is [indirectly called from here](https://github.com/google/syzkaller/blob/master/pkg/bisect/bisect.go#L123) before every bisection attempt.

My initial idea was to attach to [`newLinux`](https://github.com/google/syzkaller/blob/master/pkg/vcs/linux.go#L33) instead, but Dmitry suggested that this could be problematic, as we recycle the repository between tests and hence may not re-fetch this.

Reply:
Actually now that we write about this I remember another edge case. We might be in a gvisor repo here, this could actually break things. Let me think how I can fix this for gvisor.

Reply:
IIRC we don't do bisections for gVisor and, given the heavy reliance of the code on Linux upstream repo tags and Linux kernel configs in `linux.go`, it doesn't seem like it was already written with gVisor in mind. Looks like we should just invoke `newGit` in [`vcs.NewRepo`](https://github.com/google/syzkaller/blob/master/pkg/vcs/vcs.go#L175) for gVisor and let `linux.go` be focused on kernels that are really upstream Linux kernel-based.